LanguageFeatures/Patterns/Exhaustiveness/exhaustiveness_map_A01_t01.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Switch statements and expressions with map as a matched type can
+/// never be exhaustive
+///
+/// @description Check that map pattern cannot be exhaustive
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+String test1(Map<bool, bool> m) =>
+  switch (m) {
+//^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+    {true: true} => "case-1",
+    {true: false} => "case-2",
+    {false: true} => "case-3",
+    {false: false} => "case-4",
```

Comment:
OK, this is not an exhaustive list of cases.

But you could say that we didn't even try: An empty map will obviously not match any of these. A map that only has `true` as a key or `false` as a key should be matched, and so should a map that has both.

The problem is that we can't specify the empty map without using a guard, and a guarded case is always ignored when we compute exhaustiveness.

So you might say that it is trivially true that a map pattern cannot be exhaustive, because it will never specify the empty map.

Hmm, what about `Map<bool, bool>(length: <= 0)`? That one doesn't have a guard, so it should be accessible for the exhaustiveness algorithm, but I don't think we can expect the algorithm to take that into account. Still, we could include `Map<bool, bool>(length: <= 0) => "case-0",` just to try it out.

Reply:
I think it makes sense to add `Map<bool, bool>(length: <= 0)` as a test that will proof that lifted space union for a relational pattern is an empty space union. Thanx!